### PR TITLE
new users fixed

### LINKED ISSUE #92: Creating new user raises exception
user_params:
`<ActionController::Parameters {"first_name"=>"", "last_name"=>"", "spire"=>"", "email"=>"", "phone"=>"", "reminders_enabled"=>"1", "change_notifications_enabled"=>"1", "rosters"=>["", "2"], "membership"=><ActionController::Parameters {"admin"=>"0"} permitted: true>} permitted: true>`

Exception:
Roster(#70242086014780) expected, got "" which is an instance of String(#70242013708040)

### PR DESCRIPTION
**Pull request by avacmeyer**
#92 New users can be made now.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -12,12 +12,13 @@ class UsersController < ApplicationController
 
   def create
     user_params = params.require(:user).permit(*WHITELISTED_ATTRIBUTES)
-    user = User.new user_params
-    user.rosters << @roster
-    if user.save
-      confirm_change(user)
+    membership_params = user_params[:membership]
+    user_params = parse_roster_ids(user_params.except(:membership))
+    @user = User.new(user_params)
```
Comment (Anbranin) [inline app/controllers/users_controller.rb]: Why does `@user` need to be an instance variable?

**Comment by Anbranin** [comment]
Also, run rubocop

File: app/models/user.rb [hunk 2]
```diff
@@ -36,4 +36,5 @@ def admin?
   def membership_in(roster)
     memberships.find_by(roster: roster)
   end
+
```
Comment (dfaulken) [inline app/models/user.rb]: I think Rubocop might object to the presence of this line.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (avacmeyer) [reply app/controllers/users_controller.rb]: Because we called a method that uses the instance variable.

**Comment by Anbranin** [comment]
Are the test failures not a concern?

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (Anbranin) [reply app/controllers/users_controller.rb]: Should be a local variable.

Comment (dfaulken) [reply app/controllers/users_controller.rb]: No, @avacmeyer is correct. [The update membership method](https://github.com/umts/screaming-dinosaur/blob/ef61b663e6a61f224bc3d8888338988b2ce55264/app/controllers/users_controller.rb#L68-L74) uses an instance variable.

Comment (Anbranin) [reply app/controllers/users_controller.rb]: oh weird ok

**Pull request merged**